### DISCUSSION TITLE
How can I optimize my query?

### DISCUSSION THREAD

**Discussion by baptisteArno** [prisma/prisma#16419] [Q&A]
Hi,

I'm running out of idea on how to optimize my query:

```ts
  await prisma.result.findMany({
    take: 50,
    where: {
      typebot: {
        OR: [
          {
            id:
              typeof typebotIds === 'string' ? typebotIds : { in: typebotIds },
            collaborators: {
              some: {
                userId: user.id,
              },
            },
          },
          {
            id:
              typeof typebotIds === 'string' ? typebotIds : { in: typebotIds },
            workspace: {
              members: {
                some: {
                  userId: user.id,
                  role: { not: WorkspaceRole.GUEST },
                },
              },
            },
          },
        ],
      },
      answers: { some: {} },
    },
    orderBy: {
      createdAt: 'desc',
    },
    include: { answers: true },
  })
```

The schema: https://github.com/baptisteArno/typebot.io/blob/main/packages/db/prisma/schema.prisma

And here is the EXPLAIN result: https://explain.depesz.com/s/cjP2#html

How can I know what can be improved ?

**Comment by pantharshit00** [comment]
I have converted this into a performance issue: https://github.com/prisma/prisma/issues/16450

Please follow it there now. 

Also, can you please try enable [tracing](https://www.prisma.io/docs/concepts/components/prisma-client/opentelemetry-tracing) and see what exactly is the slowest link when you running this query. You can add information to the linked issue afterwards.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
